packages/dai-plugin-mcd/src/SavingsService.js

```diff
@@ -59,25 +59,25 @@ export default class SavingsService extends PublicService {
   }
 
   async balanceOf(guy) {
-    const slice = new BigNumber(await this._pot.pie(guy)).div(WAD);
-    const totalPie = new BigNumber(await this._pot.Pie()).div(WAD);
-
-    const portion = totalPie.eq(0) ? totalPie : slice.div(totalPie);
-
-    const daiInPot = new BigNumber(
-      await this.get('smartContract')
-        .getContract('MCD_VAT')
-        .dai(this._pot.address)
-    ).div(RAD);
-
-    return MDAI(daiInPot.times(portion));
+    const slice = new BigNumber(await this._pot.pie(guy));
+    const chi = await this.chi();
+    return MDAI(
+      slice
+        .times(chi)
+        .div(WAD)
+        .dp(18)
```

Comment:
this can round up if the sub-wei amount is greater than 0.5 wei. after reading the discussion in the smart contracts chat, i suppose this is ok and if you attempt to withdraw this amount you'll just end up with 1 wei less. perhaps the thing to do to have a deterministic test is to assert "either the value is equal to what's expected or smaller by exactly 1e-18" @b-pmcg